src/eccentricity.js

```diff
@@ -0,0 +1,70 @@
+module.exports = eccentricity;
+
+/**
+ * The eccentricity centrality of a node is the greatest distance between that node and
+ * any other node in the network. 
+ */
+function eccentricity(graph, oriented) {
+  var Q = [];
+  // distance from source
+  var dist = Object.create(null);
+
+  var currentNode;
+  var centrality = Object.create(null);
+
+  graph.forEachNode(setCentralityToZero);
+  graph.forEachNode(calculateCentrality);
+
+  return centrality;
+
+  function setCentralityToZero(node) {
+    centrality[node.id] = 0;
+  }
+
+  function calculateCentrality(node) {
+    currentNode = node.id;
+    singleSourceShortestPath(currentNode);
+    accumulate();
+  }
+
+  function accumulate() {
+    // Add all distances for node to array, excluding -1s
+    var distances = Object.keys(dist).map(function(key) {return dist[key]}).filter(function(val){return val !== -1});

```

Comment:
I'd rewrite this a little bit to reduce extra memory consumption and avoid two loops. Not that it matters in big O terms, but in practice could give faster computation times:

``` js
var maxDist = Number.NEGATIVE_INFINITY;
Object.keys(dist).forEach(function (key) {
  var val = dist[key];
  if (maxDist < val) maxDist = val;
})

centrality[currentNode] = maxDist;
```